src/pipeline/src/etl/processor/dissect.rs

```diff
@@ -543,6 +542,27 @@ impl DissectProcessor {
 
         Err("No matching pattern found".to_string())
     }
+
+    fn update_output_keys(&mut self) {
+        for fields in self.fields.iter_mut() {
+            for pattern in self.patterns.iter() {
+                for part in pattern.iter() {
+                    if let Part::Name(name) = part {
+                        if !name.is_empty()
+                            && !name
+                                .start_modifier
+                                .as_ref()
+                                .is_some_and(|x| *x == StartModifier::NamedSkip)
+                        {
+                            fields
+                                .output_fields_index_mapping
+                                .insert(name.to_string(), 0);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Nonblocking, code style: Too many nested layers, can it be modified like this:
```suggestion
        let output_keys = self
            .patterns
            .iter()
            .flat_map(|pattern| pattern.iter())
            .filter_map(|part| {
                if let Part::Name(name) = part {
                    if !name.is_empty()
                        && !name
                            .start_modifier
                            .as_ref()
                            .is_some_and(|x| *x == StartModifier::NamedSkip)
                    {
                        Some(name.to_string())
                    } else {
                        None
                    }
                } else {
                    None
                }
            })
            .collect::<HashSet<_>>();

        for field in self.fields.iter_mut() {
            for key in &output_keys {
                field.output_fields_index_mapping.insert(key.clone(), 0);
            }
        }
```

In addition, this modification also avoids multiple repeated iterations of the `pattern`.